common/scala/src/main/scala/whisk/common/Config.scala

```diff
@@ -42,113 +42,120 @@ import scala.util.Try
  * @param optionalProperties a Set of optional properties which may or not be defined.
  * @param env an optional environment to read from (defaults to sys.env).
  */
-class Config(
-    requiredProperties: Map[String, String],
-    optionalProperties: Set[String] = Set())(
-        env: Map[String, String] = sys.env)(
-            implicit logging: Logging) {
-
-    private val settings = getProperties().toMap.filter {
-        case (k, v) => requiredProperties.contains(k) ||
-            (optionalProperties.contains(k) && v != null)
-    }
+class Config(requiredProperties: Map[String, String], optionalProperties: Set[String] = Set())(
+    env: Map[String, String] = sys.env)(implicit logging: Logging) {
 
-    lazy val isValid: Boolean = Config.validateProperties(requiredProperties, settings)
-
-    /**
-     * Gets value for key if it exists else the empty string.
-     * The value of the override key will instead be returned if its value is present in the map.
-     *
-     * @param key to lookup
-     * @param overrideKey the property whose value will be returned if the map contains the override key.
-     * @return value for the key or the empty string if the key does not have a value/does not exist
-     */
-    def apply(key: String, overrideKey: String = ""): String = {
-        Try(settings(overrideKey)).orElse(Try(settings(key))).getOrElse("")
-    }
+  private val settings = getProperties().toMap.filter {
+    case (k, v) =>
+      requiredProperties.contains(k) ||
+        (optionalProperties.contains(k) && v != null)
+  }
 
-    /**
-     * Returns the value of a given key.
-     *
-     * @param key the property that has to be returned.
-     */
-    def getProperty(key: String): String = {
-        this(key)
-    }
+  lazy val isValid: Boolean =
+    Config.validateProperties(requiredProperties, settings)
```

Comment:
This line seems to break a little early - statement readability would be improved in a single line.